### PR TITLE
Send messages to NSQ to support the status api

### PR DESCRIPTION
**Pull request by msluther**
Sending along various payloads to a new NSQ topic that will be consumed by the warehouse.ai status API.  These payloads include chatty `event` messages that describe whats going on internally, and also `queued`, `ignored`, & `error` events that will more directly affect the status api.

### CODE REVIEW

**Comment by msluther** [comment]
Travis failure is `node@6` only and due to usage of the spread operator in the test code.  Will submit a fix.

File: lib/construct/progress.js [hunk 1]
```diff
@@ -160,9 +183,58 @@ Progress.prototype.write = function write(data, id) {
...
+
+  const { name, env, version, type: buildType } = this.metadata || {};
+  const freshSpec = {
+    eventType: 'event',
+    name,
+    env,
+    version,
+    locale: data.locale,
+    buildType,
+    message: JSON.stringify(data)
```
Comment (jcrugzz) [inline lib/construct/progress.js]: I feel this message should be human readable and be a meaningful description that a consumer could grok.

**Review by jcrugzz — CHANGES_REQUESTED** [review]
This seems good, id just also like to see some documentation of the status data-structures, especially with defining the various `eventType`s which will be necessary for the status-api to understand. This will then feed into the status-api docs.

**Comment by jcrugzz** [comment]
As I also mentioned to you in chat, we can allow failures for node@10 for now using the `.travis.yml` config https://github.com/winstonjs/winston/blob/master/.travis.yml#L16-L18.

**Review by jcrugzz — CHANGES_REQUESTED** [review]
Another note I want to make @msluther is I think we should send some events around the `npm install` bit which currently doesn't have access to the progress stream. I think its worth the slight refactor to be able to indicate status before `npm install` and after to indicate specific success/failures around that.

File: lib/construct/progress.js (hunk 1 above)
Comment (msluther) [reply lib/construct/progress.js]: Fair point.  data.message seems like the right thing here, its already intentionally human readable, but i'll include data.progress as well and interpolate them.

**Review by jcrugzz — APPROVED** [review]

File: lib/construct/progress.js [hunk 2]
```diff
@@ -135,13 +153,17 @@ Progress.prototype.ignore = function ignore() {
...
  * @param {String} id Optional unique id v4.
+ * @param {Object} [options] Options object
+ * @param {Boolean} [options.skipNsq] True if no NSQ event should be written, typically used for end-states
  * @returns {Progress} fluent interface
  * @api public
  */
-Progress.prototype.write = function write(data, id) {
-  if (typeof data !== 'object') {
+Progress.prototype.write = function write(data, id, options = {}) {
+  if (typeof (data) === 'string') {
```
Comment (jcrugzz) [inline lib/construct/progress.js]: no need for parens after `typeof`

**Comment by indexzero** [comment]
Post humous 👍 💯 ... nice work @msluther! <3

**Pull request merged**